cosmwasm-plugin/main.go

```diff
@@ -0,0 +1,271 @@
+				// Examples of adding subcommands:
+				Commands: []plugin.Command{
+					{
+						Use:               "add",
+						Short:             "Add a CosmWasm module to the chain",
+						Long:              "This command will install dependencies, create new files, and modify existing files to add a CosmWasm module to the chain.",
+						PlaceCommandUnder: "ignite cosmwasm-plugin",
+					},
+				},
+			},
+		},
+		// Add hooks here
+		Hooks:      []plugin.Hook{},
+		SharedHost: false,
+	}, nil
+}
+
+func (p) Execute(cmd plugin.ExecutedCommand) error {
+
+	fmt.Printf("Hello I'm the cosmwasm-plugin\n")
+
+	// This is how the plugin can access the chain:
+	// c, err := getChain(cmd)
+
+	// According to the number of declared commands, you may need a switch:
+
+	switch cmd.Use {
+	case "add":
+		return handleAddCommand()
+	default:
+		return fmt.Errorf("unknown command: %s", cmd.Use)
+	}
+
+	return nil
+}
+
+func handleAddCommand() error {
+	fmt.Println("Adding stuff...")
+
+	err := installDependencies()
+	if err != nil {
+		return err
+	}
+
+	err = createNewFiles()
+	if err != nil {
+		return err
+	}
+
+	//err = modifyExistingFiles()
+	//if err != nil {return err}
+
+	return nil
+}
+
+func (p) ExecuteHookPre(hook plugin.ExecutedHook) error {
+	fmt.Printf("Executing hook pre %q\n", hook.Name)
+	return nil
+}
+
+func (p) ExecuteHookPost(hook plugin.ExecutedHook) error {
+	fmt.Printf("Executing hook post %q\n", hook.Name)
+	return nil
+}
+
+func (p) ExecuteHookCleanUp(hook plugin.ExecutedHook) error {
+	fmt.Printf("Executing hook cleanup %q\n", hook.Name)
+	return nil
+}
+
+// For future use
+func getChain(cmd plugin.ExecutedCommand, chainOption ...chain.Option) (*chain.Chain, error) {
+	var (
+		home, _ = cmd.Flags().GetString("home")
+		path, _ = cmd.Flags().GetString("path")
+	)
+	if home != "" {
+		chainOption = append(chainOption, chain.HomePath(home))
+	}
+	absPath, err := filepath.Abs(path)
+	if err != nil {
+		return nil, err
+	}
+	return chain.New(absPath, chainOption...)
+}
+
+func installDependencies() error {
+	cmd := exec.Command("go", "get", "github.com/CosmWasm/wasmd/x/wasm@v0.41.0")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
+	if err != nil {
+		return fmt.Errorf("failed to install dependency: %w", err)
+	}
+	return nil
+}
+
+func replaceWordsInFile(filePath, targetWord, replacement string) error {
+	// Read content from the file
+	content, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return err
+	}
+
+	// Perform the replacement
+	modifiedContent := strings.ReplaceAll(string(content), targetWord, replacement)
+
+	// Write the modified content back to the file
+	err = ioutil.WriteFile(filePath, []byte(modifiedContent), 0644)
+	if err != nil {
+		return err
+	}
+
+	//fmt.Printf("Replaced planet with %s\n", replacement)
+
+	return nil
+}
+
+func createFile(inputFilename, outputDir, outputFilename string) error {
+	sourcePath := filepath.Join("..", "cosmwasm-plugin", "wasm-wiring", inputFilename)
```

Comment:
Noted and implemented with latest [commit](https://github.com/ignite/plugins/pull/2/commits/9ae9622bd2527f42c14bf4bca81ea85eac4d55a6). It would be super useful for me.